src/java.base/share/classes/jdk/internal/util/StaticProperty.java

```diff
@@ -79,6 +95,22 @@ private StaticProperty() {}
         OS_NAME = getProperty(props, "os.name");
         OS_ARCH = getProperty(props, "os.arch");
         OS_VERSION = getProperty(props, "os.version");
+        USER_LANGUAGE = getProperty(props, "user.language", "en");
+        USER_LANGUAGE_DISPLAY = getProperty(props, "user.language.display", USER_LANGUAGE);
+        USER_LANGUAGE_FORMAT = getProperty(props, "user.language.format", USER_LANGUAGE);
+        USER_SCRIPT = getProperty(props, "user.script", "");
+        USER_SCRIPT_DISPLAY = getProperty(props, "user.script.display", USER_SCRIPT);
+        USER_SCRIPT_FORMAT = getProperty(props, "user.script.format", USER_SCRIPT);
+        USER_COUNTRY = getProperty(props, "user.country", "");
+        USER_COUNTRY_DISPLAY = getProperty(props, "user.country.display", USER_COUNTRY);
+        USER_COUNTRY_FORMAT = getProperty(props, "user.country.format", USER_COUNTRY);
+        USER_VARIANT = getProperty(props, "user.variant", "");
+        USER_VARIANT_DISPLAY = getProperty(props, "user.variant.display", USER_VARIANT);
+        USER_VARIANT_FORMAT = getProperty(props, "user.variant.format", USER_VARIANT);
+        USER_EXTENSIONS = getProperty(props, "user.extensions", "");
+        USER_EXTENSIONS_DISPLAY = getProperty(props, "user.extensions.display", USER_EXTENSIONS);
+        USER_EXTENSIONS_FORMAT = getProperty(props, "user.extensions.format", USER_EXTENSIONS);
+        USER_REGION = getProperty(props, "user.region", "");
```

Comment:
Computing the defaults for these properties should be in the Locale class, close to the initialization of the default locale. Splitting the responsibility across files makes it harder to follow what happens where/when.

Reply:
If I am not mistaken, these assignments should stay here in `StaticProperty`, as that is the whole purpose of this change. Depending on `Locale` class loading time might be fragile, and this change guarantees the properties are initialized and frozen in `System.initPhase1()`